divviup/src/test/java/org/divviup/android/JanusIntegrationTest.java

```diff
@@ -0,0 +1,439 @@
+                    vdaf,
+                    collectorAuthToken,
+                    queryType
+            );
+
+            // Provision task into leader
+            leaderApi.addTask(
+                    taskId,
+                    "leader",
+                    leaderUriDocker,
+                    helperUriDocker,
+                    vdaf,
+                    aggregatorAuthToken,
+                    collectorAuthToken,
+                    encodedVdafVerifyKey,
+                    maxBatchQueryCount,
+                    batchSize,
+                    batchSize,
+                    encodedCollectorHpkeConfig,
+                    taskExpirationTimestamp,
+                    queryType
+            );
+
+            // Provision task into helper
+            helperApi.addTask(
+                    taskId,
+                    "helper",
+                    leaderUriDocker,
+                    helperUriDocker,
+                    vdaf,
+                    aggregatorAuthToken,
+                    null,
+                    encodedVdafVerifyKey,
+                    maxBatchQueryCount,
+                    batchSize,
+                    batchSize,
+                    encodedCollectorHpkeConfig,
+                    taskExpirationTimestamp,
+                    queryType
+            );
+
+            // Send measurements
+            Client<M> client = clientConstructor.construct(leaderUriHost, helperUriHost, taskId);
+            for (M measurement : measurements) {
+                client.sendMeasurement(measurement);
+            }
+
+            // Start collection
+            ObjectNode query = JsonNodeFactory.instance.objectNode();
+            query.set("type", JsonNodeFactory.instance.numberNode(queryType));
+            query.set("subtype", JsonNodeFactory.instance.numberNode(1)); // current_batch
+
+            result = null;
+            for (int startAttempt = 0; startAttempt < 5; startAttempt++) {
+                try {
+                    String handle = collectorApi.startCollection(taskId, new byte[0], query);
+
+                    for (int pollAttempt = 0; pollAttempt < 30; pollAttempt++) {
+                        result = collectorApi.pollCollection(handle);
+                        if (result != null && result != NullNode.getInstance()) {
+                            break;
+                        }
+                        Thread.sleep(1000);
+                    }
+                    break;
+                } catch (InteropApiException e) {
+                    Thread.sleep(1000);
+                }
+            }
+        } finally {
+            propagateLogs(leader, "leader");
+            propagateLogs(helper, "helper");
+            propagateLogs(collector, "collector");
+        }
+
+        assertNotNull(result);
+        assertEquals(expectedResult, result);
+    }
+
+    private static TaskId randomTaskId() {
+        byte[] bytes = new byte[32];
+        new Random().nextBytes(bytes);
```

Comment:
Possibly nit, can leave it if you don't think it's pertinent: `java.util.Random` [is not a CSPRNG](https://docs.oracle.com/javase/8/docs/api/java/util/Random.html).

For practical usage in these tests it probably doesn't matter, but I would be worried about someone reading these tests for reference on how to use the library, and copying over usage of `java.util.Random`.